crates/rattler_solve/src/resolvo/conda_util.rs

```diff
@@ -1,17 +1,344 @@
+        let a_has_tracked_features = !a_record.track_features().is_empty();
+        let b_has_tracked_features = !b_record.track_features().is_empty();
+        match a_has_tracked_features.cmp(&b_has_tracked_features) {
+            Ordering::Less => return Ordering::Less,
+            Ordering::Greater => return Ordering::Greater,
+            Ordering::Equal => {}
+        };
+
+        // Otherwise, select the variant with the highest version
+        match (self.strategy, a_record.version().cmp(b_record.version())) {
+            (CompareStrategy::Default, Ordering::Greater)
+            | (CompareStrategy::LowestVersion, Ordering::Less) => return Ordering::Less,
+            (CompareStrategy::Default, Ordering::Less)
+            | (CompareStrategy::LowestVersion, Ordering::Greater) => return Ordering::Greater,
+            (_, Ordering::Equal) => {}
+        };
+
+        // Otherwise, select the variant with the highest build number
+        match a_record.build_number().cmp(&b_record.build_number()) {
+            Ordering::Less => return Ordering::Greater,
+            Ordering::Greater => return Ordering::Less,
+            Ordering::Equal => return Ordering::Equal,
+        };
+    }
+
+    pub fn sort_by_dependencies(
+        &self,
+        solvables: &mut [SolvableId],
+        version_cache: &mut HashMap<VersionSetId, Option<(Version, bool)>>,
+    ) {
+        let mut start = 0usize;
+        let entire_len = solvables.len();
+        while start < entire_len {
+            let mut end = start + 1;
+
+            // Find the range of solvables with the same: version, build, tracked features
+            while end < entire_len
+                && self.simple_compare(solvables[start], solvables[end]) == Ordering::Equal
+            {
+                end += 1;
+            }
+
+            // Take the sub list of solvables
+            let sub = &mut solvables[start..end];
+            if sub.len() > 1 {
+                // Sort the sub list of solvables by the highest version of the dependencies
+                self.sort_by_highest_version(sub, version_cache);
+            }
+
+            start = end;
+        }
+    }
+
+    /// Sorts the solvables by the highest version of the dependencies shared by the solvables.
+    /// what this function does is:
+    /// 1. Find the first unsorted solvable in the list
+    /// 2. Get the dependencies for each solvable
+    /// 3. Get the known dependencies for each solvable, filter out the unknown dependencies
+    /// 4. Retain the dependencies that are shared by all the solvables
+    /// 5. Create a max vector which is the maximum version of each of the shared dependencies, per dependency name
+    /// 6. Calculate a total score  by counting the position of the solvable in the list with sorted dependencies
+    /// 7. Sort by the total score and use timestamp of the record as a tie breaker
+    fn sort_by_highest_version(
```

Comment:
Maybe we should rename this function `sort_by_highest_dependency_versions`?

Reply:
Yes will change this later.